database/engine/metadata_log/metadatalogapi.c

```diff
@@ -3,460 +3,479 @@
-void metalog_delete_dimension_by_uuid(struct metalog_instance *ctx, uuid_t *metric_uuid)
-{
-    RRDDIM *rd;
-    RRDSET *st;
-    RRDHOST *host;
-    uint8_t empty_chart;
-
-    rd = metalog_get_dimension_from_uuid(ctx, metric_uuid);
-    if (!rd) { /* in the case of legacy UUID convert to multihost and try again */
-        uuid_t multihost_uuid;
-
-        rrdeng_convert_legacy_uuid_to_multihost(ctx->rrdeng_ctx->machine_guid, metric_uuid, &multihost_uuid);
-        rd = metalog_get_dimension_from_uuid(ctx, &multihost_uuid);
-    }
-    if(!rd) {
-        info("Rotated unknown archived metric.");
-        return;
-    }
-    st = rd->rrdset;
-    host = st->rrdhost;
-
-    /* In case there are active metrics in a different database engine do not delete the dimension object */
-    if (unlikely(host->rrd_memory_mode != RRD_MEMORY_MODE_DBENGINE))
-        return;
-
-    /* Since the metric has no writer it will not be commited to the metadata log by rrddim_free_custom().
-     * It must be commited explicitly before calling rrddim_free_custom(). */
-    metalog_commit_delete_dimension(rd);
-
-    rrdset_wrlock(st);
-    rrddim_free_custom(st, rd, 1);
-    empty_chart = (NULL == st->dimensions);
-    rrdset_unlock(st);
-
-    if (empty_chart) {
-        rrdhost_wrlock(host);
-        rrdset_rdlock(st);
-        rrdset_delete_custom(st, 1);
-        rrdset_unlock(st);
-        rrdset_free(st);
-        rrdhost_unlock(host);
-    }
-}
-
-void metalog_print_dimension_by_uuid(struct metalog_instance *ctx, uuid_t *metric_uuid)
-{
-    RRDDIM *rd;
-    RRDSET *st;
-    RRDHOST *host;
-
-    if (!ctx || !ctx->initialized)
-        return;
-
-    rd = metalog_get_dimension_from_uuid(ctx, metric_uuid);
-    if (!rd) { /* in the case of legacy UUID convert to multihost and try again */
-        uuid_t multihost_uuid;
-
-        rrdeng_convert_legacy_uuid_to_multihost(ctx->rrdeng_ctx->machine_guid, metric_uuid, &multihost_uuid);
-        rd = metalog_get_dimension_from_uuid(ctx, &multihost_uuid);
-    }
-    if(!rd) {
-        error_with_guid(metric_uuid, "GUID not found, unknown metric.");
-        return;
-    }
-    st = rd->rrdset;
-    host = st->rrdhost;
-
-    error_with_guid(metric_uuid, "Host - Chart - Dimension are the below:");
-    error("%s %s %s.", host->hostname, st->id, rd->id);
-
-    if (unlikely(host->rrd_memory_mode != RRD_MEMORY_MODE_DBENGINE))
-        error_with_guid(metric_uuid, "UUID does not belong to RRD_MEMORY_MODE_DBENGINE.");
-}
+//void metalog_delete_dimension_by_uuid(struct metalog_instance *ctx, uuid_t *metric_uuid)
+//{
+//    UNUSED(ctx);
+//    RRDDIM *rd;
+//    RRDSET *st;
+//    RRDHOST *host;
+//    uint8_t empty_chart;
+//    rd = metalog_get_dimension_from_uuid(ctx, metric_uuid);
+//    if (!rd) { /* in the case of legacy UUID convert to multihost and try again */
+//        uuid_t multihost_uuid;
+//
+//        rrdeng_convert_legacy_uuid_to_multihost(ctx->rrdeng_ctx->machine_guid, metric_uuid, &multihost_uuid);
```

Comment:
This function is skipped, but it is essential otherwise the dbengine UUIDs from different legacy instances collide.